### PR TITLE
repair: throw abort_requested_exception when abort is requested

### PR DESCRIPTION
**Pull request by Deexie**
If abort is requsted during bootstrap then a node should exit normally. To achieve so, abort_requested_exception should be thrown as main handles it gracefully.

In data_sync_repair_task_impl::run exceptions from all shards are wrapped together into std::runtime_exception and so they aren't handled as they are supposed to.

Throw abort_requested_exception when shutdown was requested. Throw abort_requested_exception also if repair::task_manager_module::is_aborted, so that force_terminate_all_repair_sessions acts the same regardless the state of the repair.

To maintain consistency do the same for user_requested_repair_task_impl.

Fixes: #15710.

### CODE REVIEW

File: repair/repair.cc [hunk 1]
```diff
@@ -1304,8 +1304,9 @@ future<> repair::user_requested_repair_task_impl::run() {
             }
             return make_ready_future<>();
         }).get();
-    }).handle_exception([id] (std::exception_ptr ep) {
+    }).handle_exception([id, &rs] (std::exception_ptr ep) {
         rlogger.warn("repair[{}]: user-requested repair failed: {}", id.uuid(), ep);
+        rs.get_repair_module().check_in_shutdown();
```
Comment (denesb) [inline repair/repair.cc:1309]: Why is this needed?

Comment (Deexie) [reply repair/repair.cc:1309]: `check_in_shutdown` calls `check` on abort source and so if abort was requested it throws `abort_requested_exception` instead of exceptions from all shards wrapped.

Comment (Deexie) [reply repair/repair.cc:1309]: Line 1275 is reached only when repair was aborted with force_terminate_all_repair_sessions api

Comment (denesb) [reply repair/repair.cc:1309]: I think this is worth a comment, also at the other similar place in this patch. It is suprising that we are discarding the current exception for a new one, so we need to commend to explain why this is ok.

**Comment by Deexie** [comment]
- add comments to `check_in_shutdown` call

**Review by denesb — APPROVED** [review]
